### PR TITLE
fix: Interactive terminal freeze in tests

### PR DESCRIPTION
**Pull request by michaljurecko**
Slack: https://keboolaglobal.slack.com/archives/C055WKPHYSD/p1724837155893959

**Changes:**
- Fixed bug/edge-case in interactive terminal tests.
  - Added workaround for issue in `vtx10/survey` libs.

-----------

**Debug:**

Test freezes if the `VAR1` stdin and `<ansi>` terminal sequence are read at once.

image

### CODE REVIEW

File: internal/pkg/utils/testhelper/terminal/terminal_linux_macos.go [hunk 1]
```diff
@@ -208,7 +253,8 @@ func (t *tty) Close() error {
 		return errors.New("tty already closed")
 	default:
 		close(t.closed)
-		return t.file.Close()
+		err := t.file.Close()
+		return err
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/utils/testhelper/terminal/terminal_linux_macos.go:257]: yes, I will fix it

**Review by Matovidlo — APPROVED** [review]

File: internal/pkg/utils/testhelper/terminal/terminal_linux_macos.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/utils/testhelper/terminal/terminal_linux_macos.go]: https://github.com/keboola/keboola-as-code/pull/2005/commits/99683eb80b12c723e3b02511b7beb459c1bd5283
